src/main/java/ua/bookstore/online/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package ua.bookstore.online.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ua.bookstore.online.dto.shopping.cart.CartItemRequestDto;
+import ua.bookstore.online.dto.shopping.cart.CartItemResponseDto;
+import ua.bookstore.online.dto.shopping.cart.QuantityDto;
+import ua.bookstore.online.exception.EntityNotFoundException;
+import ua.bookstore.online.mapper.CartItemMapper;
+import ua.bookstore.online.model.Book;
+import ua.bookstore.online.model.CartItem;
+import ua.bookstore.online.model.ShoppingCart;
+import ua.bookstore.online.repository.book.BookRepository;
+import ua.bookstore.online.repository.shopping.cart.item.CartItemRepository;
+import ua.bookstore.online.service.CartItemService;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public CartItemResponseDto add(CartItemRequestDto requestDto, ShoppingCart shoppingCart) {
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(
+                () -> new EntityNotFoundException("Book not found by id " + requestDto.bookId()));
```

Comment:
I believe you can use book service, to not duplicate code

Reply:
In this case I should change book service or book mapper or use something like this:
```suggestion
        Long bookId = requestDto.bookId();
        bookService.getById(bookId); //to check if exist.
        Book book = new Book();
        book.setId(bookId);
```

Reply:
I don`t understand why the code that you write is required. That goal is to find book by id, and there is a method in bookService that is responsible for that and encapsulates orElseThrow logic